### PR TITLE
Add converter from Yosys JSON to FPGA interchange logical netlist.

### PR DESCRIPTION
**Pull request by litghost**

### CODE REVIEW

File: fpga_interchange/yosys_json.py [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+    """
+    if not upto:
+        for idx, bit in enumerate(bits):
+            yield idx + offset, bit
+    else:
+        for idx, bit in enumerate(bits):
+            yield offset + (len(bits) - 1) - idx, bit
+
+
+# FIXME: These constants should likely be from the device database.
```
Comment (gatecat) [inline fpga_interchange/yosys_json.py:75]: Admittedly I don't have much context here, but have you thought about having constants as a special case in the schema? Requiring a netlist transformation to know about device-specific cells seems a bit suboptimal to me.

File: fpga_interchange/yosys_json.py [hunk 2]
```diff
@@ -0,0 +1,417 @@
...
+
+        offset = net_data.get('offset', 0)
+        upto = net_data.get('upto', 0)
+        if is_bus(net_data['bits'], offset, upto):
+            for bit_index, bit in interp_yosys_net(net_data['bits'], offset,
+                                                   upto):
+                name = '{}[{}]'.format(net_name, bit_index)
+                cell.add_net(name, property_map)
+
+                if bit == '0':
```
Comment (gatecat) [inline fpga_interchange/yosys_json.py:146]: 'x' and 'z' bits can also occur, in this context I think converting these to disconnected ports for both should be OK

File: fpga_interchange/yosys_json.py (hunk 1 above)
Comment (litghost) [reply fpga_interchange/yosys_json.py:75]: This is a gap that needs to be filled, both for this and the pin map case.

File: fpga_interchange/yosys_json.py (hunk 2 above)
Comment (litghost) [reply fpga_interchange/yosys_json.py:146]: `'x'` and `'z'` bits need to be handled prior to the FPGA interchange logical netlist format.  For example my murax yosys script is currently:

```
yosys -import

read_verilog Murax.v
read_verilog basys3_toplevel.v

synth_xilinx -nolutram -nowidelut -nosrl -nocarry -nodsp

# opt_expr -undriven makes sure all nets are driven, if only by the $undef
# net.
opt_expr -undriven
opt_clean

setundef -zero -params

write_json murax.json
```

So while `'x'` and `'z'` bits legal in Yosys Netlist JSON, they are not legal as input to the FPGA interchange.

**Pull request merged**